tests/integration/test_tf_symbolic.py

```diff
@@ -565,7 +565,13 @@ def test_eval_true_state_all_fock_probs(self, setup_eng, cutoff, batch_size, tol
             Dgate(-ALPHA) | q[1]
 
         state = eng.run(prog).state
-        probs = state.all_fock_probs(eval=True).flatten()
+
+        if eng.backend_name == 'tf':
+            import tensorflow as tf
+            probs = tf.reshape(state.all_fock_probs(eval=True), [-1])
```

Comment:
I think currently the backend API measurement methods do not return consistent types anyway (#300 converts the results to a consistent type only later, in `Result.__init__`). I think it'd be better if all the backend API measurement methods returned an `array` of a consistent shape. This was one of the changes implemented in #111, along with the multishot capability.

Reply:
>  I think it'd be better if all the backend API measurement methods returned an array of a consistent shape. This was one of the changes implemented in #111, along with the multishot capability.

I agree. Are the changes in #111 portable to the latest master?

Reply:
I can fix them so they are, but `tf2` should probably be merged to master first.

Reply:
It keeps getting forgotten in all these discussions, but we specifically want the tf backend to return tf.tensors. If we convert these to arrays, then we are breaking the computational graph, which may negate the point of using tf backend to begin with

Reply:
I guess returning tf.Tensors could be a special property of the tf backend, but the Tensors should always have the same shape (and similar dtype) as the arrays returned by the other backends. Previously we had the optional `eval` argument that decided whether the tf backend returned Tensors or arrays, right?

Reply:
Oh, I'm not actually sure. I thought `eval` was deprecated now that we support eager mode.

Reply:
That's right @smite, TF backend should return tensor objects whose data has the same shape/dtype as the numpy arrays returned by the other backends.

`eval` keyword argument is no longer necessary, since TF2 has moved to eager mode and deprecated all the logic around `session`s. Now there's less need to distinguish unevaluated vs evaluated tensors. However, we still need to return tf eager tensors, not numpy arrays, in order to properly support backpropagation